pkg/flowcli/services/keys.go

```diff
@@ -78,16 +82,45 @@ func (k *Keys) Generate(inputSeed string, signatureAlgo string) (crypto.PrivateK
 	return privateKey, nil
 }
 
-func (k *Keys) Decode(publicKey string) (*flow.AccountKey, error) {
+// Decode encoded public key using supported encoding type
+func (k *Keys) Decode(encoded string, encoding string) (*flow.AccountKey, error) {
+	if strings.ToLower(encoding) == PEM {
+		return decodePEM(encoded)
+	} else if strings.ToLower(encoding) == RLP {
+		return decodeRLP(encoded)
+	}
+
+	return nil, fmt.Errorf("encoding type not supported. Valid encodings: RLP, PEM")
+}
+
+func decodeRLP(publicKey string) (*flow.AccountKey, error) {
 	publicKeyBytes, err := hex.DecodeString(publicKey)
 	if err != nil {
 		return nil, fmt.Errorf("failed to decode public key: %v", err)
 	}
 
 	accountKey, err := flow.DecodeAccountKey(publicKeyBytes)
 	if err != nil {
-		return nil, fmt.Errorf("failed to decode private key bytes: %v", err)
+		return nil, fmt.Errorf("failed to decode: %v", err)
 	}
 
 	return accountKey, nil
 }
+
+func decodePEM(publicKeyPath string) (*flow.AccountKey, error) {
+	fileContent, err := util.LoadFile(publicKeyPath)
+	if err != nil {
+		return nil, err
+	}
+
+	pk, err := crypto.DecodePublicKeyPEM(crypto.ECDSA_P256, string(fileContent))
```

Comment:
Is `crypto.ECDSA_P256` a valid assumption to make here?

Reply:
Good catch. I think we need to support `secp256k1` as well. Maybe add a `--sig-algo` flag to the command?

Reply:
ohh... yeah, thanks for this, I actually planned to do this and then I changed from draft to open and forgot, I will do this during the QA.